hilti/toolchain/src/compiler/validator.cc

```diff
@@ -172,6 +193,7 @@ struct VisitorPost : visitor::PreOrder, public validator::VisitorMixIn {
     }
 
     void operator()(Function* n) final {
+        checkNodeAttributes(n->nodeTag(), n->attributes(), "function");
```

Comment:
I'm wondering if could move this into a generic `operator()(Node *n)` method so that it would execute for any nodes. Main catch is probably creating the `where` for the error message ...

Reply:
Yeah, I tried and the catch was that `where`. I looked for a way to get a string saying what a node is, but I couldn't find one for all nodes - is there one that exists? I could only find some for debugging.

If not, maybe this could be done in a followup? I think having a "print node for diagnostics" would be valuable for this, but that would probably apply to every node.

Ditching the `where` would make me very sad. It was absolutely crucial for debugging, I can imagine the same would apply to users.

Reply:
> is there one that exists? I could only find some for debugging.

Yeah, nothing that would yield something nice for user display, just internal class names.

Ok to leave as is.

Reply:
> I think having a "print node for diagnostics" would be valuable for this, but that would probably apply to every node.

Maybe we could generalize `displayName()` to all nodes. (But indeed for later.)